### PR TITLE
Refactor background tasks and services out of execution units

### PR DESCRIPTION
**Pull request by mzabaluev**
# Summary
- **Categories**: `protocol-units`, `networks`.

Start on https://github.com/movementlabsxyz/movement/issues/63 by refactoring maptos-opt-executor, maptos-fin-view, and their integration in maptos-dof-execution and suzuka-full-node.

# Testing

Locally:
`cargo test -p maptos-opt-executor`
`cargo test -p maptos-fin-view`

Existing integration tests should pass in CI after the changes.

### CODE REVIEW

File: networks/suzuka/suzuka-full-node/src/lib.rs [hunk 1]
```diff
@@ -1,27 +1,7 @@
+mod da_db;
 pub mod manager;
 pub mod partial;
+mod tasks;
 
 #[cfg(test)]
 pub mod tests;
-
-pub trait SuzukaFullNode {
```
Comment (l-monninger) [inline networks/suzuka/suzuka-full-node/src/lib.rs:7]: The King is dead...

File: networks/suzuka/suzuka-full-node/src/tasks/mod.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+//! Modules to separate full node processing into actor-like tasks.
+
+pub mod exec_settle;
+pub mod tx_ingress;
```
Comment (l-monninger) [inline networks/suzuka/suzuka-full-node/src/tasks/mod.rs:4]: Prefer "transaction" over "tx."

File: protocol-units/execution/opt-executor/src/executor/initialization.rs [hunk 3]
```diff
@@ -212,4 +91,28 @@ impl Executor {
...
+
+	/// Creates instance of `Context` and the background `TransactionPipe`
+	/// task to process transactions.
+	pub fn background(
+		&self,
+		transaction_sender: mpsc::Sender<SignedTransaction>,
+	) -> (Context, TransactionPipe) {
+		// use the default signer, block executor, and mempool
+		let (mempool_client_sender, mempool_client_receiver) =
+			futures_mpsc::channel::<MempoolClientRequest>(2 ^ 16); // allow 2^16 transactions before apply backpressure given theoretical maximum TPS of 170k
```
[the remark this answers is not in the document]
Comment (l-monninger) [reply protocol-units/execution/opt-executor/src/executor/initialization.rs:103]: This still needs to be changed.

File: networks/suzuka/suzuka-full-node/src/tasks/exec_settle.rs [hunk 4]
```diff
@@ -0,0 +1,244 @@
+//! Task module to execute blocks from the DA and process settlement.
```
Comment (l-monninger) [inline networks/suzuka/suzuka-full-node/src/tasks/exec_settle.rs:1]: I'd prefer to use `execute_settle`.

**Review by l-monninger — CHANGES_REQUESTED** [review]

File: protocol-units/execution/opt-executor/src/executor/initialization.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mzabaluev) [reply protocol-units/execution/opt-executor/src/executor/initialization.rs:103]: Hmm, there was a constant for this as well. Will change.

**Review by l-monninger — APPROVED** [review]

**Pull request merged**